### DISCUSSION TITLE
Remove sigs.k8s.io/controller-runtime dependency from */api modules

### DISCUSSION THREAD

**Discussion by ash2k** [fluxcd/flux2#4782] [Proposals]
Hi! My project depends on these Flux modules:

```
	github.com/fluxcd/notification-controller/api v1.2.4
	github.com/fluxcd/pkg/apis/meta v1.3.0
	github.com/fluxcd/source-controller/api v1.2.4
```

The `api` ones have a dependency on the `sigs.k8s.io/controller-runtime/pkg/scheme` package. It brings in a lot of things I don't need an don't use. Would be nice if the `api` modules were even lighter, without the `controller-runtime` module. Thank you.

**Comment by stefanprodan** [comment]
We can't do this, our APIs are generated with controller-runtime which adds scheme.
